pkg/machines/components/diskAdd.jsx

```diff
@@ -48,10 +48,10 @@ function getAvailableTargets(vm) {
 
 function getFilteredVolumes(vmStoragePool, disks) {
     const usedDiskPaths = Object.getOwnPropertyNames(disks)
-            .filter(target => disks[target].source && disks[target].source.file)
-            .map(target => disks[target].source.file);
+            .filter(target => disks[target].source && (disks[target].source.file || disks[target].source.volume))
+            .map(target => (disks[target].source.file ? disks[target].source.file : disks[target].source.volume));
```

Comment:
This is (and already was in master) a bit inconsistent. Is `.source` always guaranteed to exist? If so, then the check in the `.filter()` call can  be dropped. If not, then its existence should also be checked in `.map()`.

Stylistic hint: `a ? a : b` is better written as `a || b` to save typing and avoid typos, like the .filter call already does.

Reply:
Source is not always present.
Usual case is the ```cdrom```, which can be removed, and in this case the ```source``` is missing.
I 'll check the existence in ```.map()``` as well.  Thanks